spring-kafka/src/main/java/org/springframework/kafka/support/KafkaHeaders.java

```diff
@@ -67,4 +68,19 @@
 	 */
 	public static final String RECEIVED_PARTITION_ID = PREFIX + "receivedPartitionId";
 
+	/**
+	 * The header for holding the {@link org.apache.kafka.common.record.TimestampType type} of timestamp.
+	 */
+	public static final String TIMESTAMP_TYPE = PREFIX + "timestampType";
+
+	/**
+	 * The header for holding the timestamp of the record.
+	 */
+	public static final String TIMESTAMP = PREFIX + "timestamp";
+
+	/**
+	 * The header for holding the timestamp of the record.
```

Comment:
I think `consumer record` as opposite to `producer record` on the `TIMESTAMP`.
Just to let end-users to avoid confusing.